R/bootstrap-model.R

```diff
@@ -0,0 +1,681 @@
+#' Create a boostrap model object from an existing model
+#'
+#' @param .mod a `bbr` model object
+#' @param .suffix a suffix for the bootstrap run directory. Will be prefixed by
+#'  the model id of `.mod`.
+#' @inheritParams copy_model_from
+#' @param remove_cov,remove_tables Logical (T/F). Optionally remove `$COVARIANCE`
+#' and `$TABLE` records respectively, allowing for notably faster run times.
+#'
+#' @seealso setup_bootstrap_run summarize_bootstrap_run
+#' @return S3 object of class `bbi_nmboot_model`.
+#' @export
+new_bootstrap_run <- function(
+    .mod,
+    .suffix = "boot",
+    .inherit_tags = TRUE,
+    .overwrite = FALSE,
+    remove_cov = TRUE,
+    remove_tables = TRUE
+){
+
+  check_model_object(.mod, NM_MOD_CLASS)
+
+  model_dir <- get_model_working_directory(.mod)
+  boot_dir <- glue("{get_model_id(.mod)}-{.suffix}")
+
+  boot_run <- copy_model_from(
+    .parent_mod = .mod,
+    .new_model = boot_dir,
+    .add_tags = "BOOTSTRAP_SUBMISSION",
+    .inherit_tags = .inherit_tags,
+    .update_model_file = TRUE,
+    .overwrite = .overwrite
+  )
+
+  boot_run[[YAML_MOD_TYPE]] <- "nmboot"
+  boot_run <- save_model_yaml(boot_run)
+
+  # Change problem statement
+  prob <- glue("Bootstrap run of model {get_model_id(.mod)}")
+  modify_prob_statement(boot_run, prob)
+
+  # Optionally remove $TABLE and $COV statements here
+  if(isTRUE(remove_cov)) remove_records(boot_run, type = "covariance")
+  if(isTRUE(remove_tables)){
+    remove_records(boot_run, type = "table")
+  }else{
+    # Update table names if present (not used, but for consistency)
+    boot_run <- update_model_id(boot_run) %>% suppressMessages()
+  }
+
+  # Return read-in model to get the updated class as part of the model object
+  return(read_model(file.path(model_dir, boot_dir)))
+}
+
+
+#' Set up a bootstrap model run.
+#'
+#' Creates a new model object and re-sampled dataset per model run.
+#'
+#' @param .boot_run a `bbi_nmboot_model` object.
+#' @param n number of model runs.
+#' @param strat_cols columns to maintain proportion for stratification
+#' @param seed a seed for sampling the data. Set to `NULL` to avoid setting.
+#' @param .overwrite logical (T/F) indicating whether or not to overwrite existing
+#'  setup for a bootstrap run.
+#'
+#' @seealso new_bootstrap_run summarize_bootstrap_run
+#'
+#' @examples
+#' \dontrun{
+#'
+#' # Setup
+#' .boot_run <- new_bootstrap_run(.mod)
+#' .boot_run <- setup_bootstrap_run(.boot_run, n = 1000, seed = 1234)
+#'
+#' # Submit
+#' submit_model(.boot_run)
+#'
+#' # Check status of runs during submission
+#' get_model_status(.boot_run)
+#' }
+#' @export
+setup_bootstrap_run <- function(
+    .boot_run,
+    n = 100,
```

Comment:
Per user feedback, change this default to `200`

Reply:
I thought that was just for the specific project template script? Do we want to confirm that value internally too?